command/agent/command.go

```diff
@@ -362,6 +399,42 @@ func (config *Config) verifyUniqueListeners() error {
 	return nil
 }
 
+// discoverEc2Hosts searches the given AWS region, returning a list of instance
+// addresses where EC2TagKey = EC2TagValue
+func (c *Config) discoverEc2Hosts() ([]string, error) {
+	config := c.EC2Discovery
+	awsConfig := &aws.Config{
+		Region:      aws.String(config.Region),
+		Credentials: credentials.NewStaticCredentials(config.AccessKeyID, config.SecretAccessKey, ""),
```

Comment:
Doesn't this preclude use of the very useful [IAM instance credentials](http://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2.html)?  While statically defining the credentials is a useful option, IAM roles remove the management overhead of rotating those keys.